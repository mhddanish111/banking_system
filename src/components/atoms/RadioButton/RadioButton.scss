@import '../../../styles/index.scss';
@import '../../../styles/mixins.scss';

@mixin absolute-position($toPx) {
  top: $toPx;
  left: $toPx;
}
.radio {
  display: flex;
  position: relative;
  align-items: baseline;
  margin-bottom: $space-xxs;
}
.radio-button {
  position: absolute;
  @include absolute-position(0);
  width: $icon-s;
  height: $icon-s;
  opacity: 0;
  + .radio-wrapper {
    + .radio-button-label {
      @include font-body;
      margin-left: $space-xxs;
      margin-bottom: unset;
    }
    .outerCircle {
      .innerCircle {
        width: $icon-xs;
        height: $icon-xs;
        border-radius: 50%;
        background: $color-gray-50;
        opacity: 0;
        margin-bottom: 0;
      }
      border: $border-dark;
      border-radius: 50%;
      width: $icon-s;
      height: $icon-s;
      display: flex;
      justify-content: center;
      align-items: center;
      background-color: $color-white;
    }
  }
  &:checked {
    + .radio-wrapper .outerCircle {
      border: $border-dark;
      .innerCircle {
        opacity: 1;
      }
    }
  }
  &:hover {
    + .radio-wrapper .outerCircle {
      border: $border-dark;
    }
  }
  &:disabled {
    + .radio-wrapper {
      + .radio-button-label {
        color: $color-disabled;
      }
      .outerCircle {
        border: $border-disabled;
      }
    }
    &:checked {
      + .radio-wrapper {
        + .radio-button-label {
          color: $color-disabled;
        }
        .outerCircle {
          border: $border-disabled;
          .innerCircle {
            opacity: 1;
            background: $color-disabled;
          }
        }
      }
    }
  }
  &:focus {
    + .radio-wrapper .outerCircle {
      border: $border-focus;
    }
  }
}

.large {
  .radio-button {
    & + .radio-wrapper + .radio-button-label {
      margin-left: $space-m;
    }
  }
}
