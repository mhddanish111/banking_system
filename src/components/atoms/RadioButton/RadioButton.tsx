import React from 'react';
import './RadioButton.scss';

type RadioButtonProps = {
  className?: string;
  value?: string;
  checked?: boolean;
  ariaLabel?: string;
  name?: string;
  id?: string;
  tabIndex?: number;
  onSelect?: any;
  children?: React.ReactNode;
  other?: any;
};
const RadioButton: React.FC<RadioButtonProps> = React.memo(
  ({
    className,
    value,
    checked,
    children,
    ariaLabel,
    name,
    id,
    tabIndex,
    onSelect,
    other,
  }) => {
    const handleChange = (e: any) => {
      if (e.target.disabled) {
        return;
      }
      if (onSelect) {
        onSelect(e.target.value);
      }
    };
    return (
      <div className={className} key={id}>
        <label htmlFor={id} className="radio">
          <input
            className="radio-button"
            checked={checked}
            value={value}
            id={id}
            name={name}
            type="radio"
            onChange={handleChange}
            tabIndex={tabIndex}
            aria-label={ariaLabel}
            aria-checked={checked}
            {...other}
          />
          <div className="radio-wrapper">
            <div className="outerCircle">
              <div className="innerCircle" />
            </div>
          </div>
          <span className="radio-button-label">{children}</span>
        </label>
      </div>
    );
  },
);

export default RadioButton;
