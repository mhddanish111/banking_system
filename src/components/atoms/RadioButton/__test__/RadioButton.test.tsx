import { shallow } from 'enzyme';
import RadioButton from '../RadioButton';

describe('<RadioButton />', () => {
  const onSelect = jest.fn();
  const props = {
    className: '',
    value: 'test',
    ariaLabel: 'radio-btn',
    checked: true,
    name: 'test',
    id: 'test',
    tabIndex: 0,
    children: <div />,
    onSelect,
  };
  const wrapper = shallow(<RadioButton {...props} />);
  it('should render the Heading Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should call onChange event', () => {
    const event = { target: { disabled: false } };
    const disabledEvent = {
      preventDefault: jest.fn(),
      target: { disabled: true },
    };
    const handleChange = jest.fn();
    wrapper
      .find('.radio-button')
      .simulate('change', event, handleChange(event));
    wrapper
      .find('.radio-button')
      .simulate('change', disabledEvent, handleChange(disabledEvent));
    expect(handleChange).toHaveBeenCalled();
  });
});
