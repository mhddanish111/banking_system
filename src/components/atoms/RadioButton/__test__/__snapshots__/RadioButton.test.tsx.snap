// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> should render the Heading Component correctly 1`] = `
<div
  className=""
  key="test"
>
  <label
    className="radio"
    htmlFor="test"
  >
    <input
      aria-checked={true}
      aria-label="radio-btn"
      checked={true}
      className="radio-button"
      id="test"
      name="test"
      onChange={[Function]}
      tabIndex={0}
      type="radio"
      value="test"
    />
    <div
      className="radio-wrapper"
    >
      <div
        className="outerCircle"
      >
        <div
          className="innerCircle"
        />
      </div>
    </div>
    <span
      className="radio-button-label"
    >
      <div />
    </span>
  </label>
</div>
`;
